File: data/code_pairs/fortran/2314.f90
module check_module
    implicit none

contains

    logical function check_is_larger(i)
        integer, intent(in) :: i
        integer, save :: last_i = 0
        logical :: is_larger

        if (i == 1) then
            last_i = 0
        endif

        is_larger = (i >= last_i) .and. ((i - last_i) <= 1)
        last_i = i

        check_is_larger = is_larger
    end function check_is_larger

end module check_module

program test_do_collapse
    use check_module
    implicit none

    ! Test check_is_larger function
    print *, 'Testing check_is_larger function'
    print *, 'check_is_larger(1):', check_is_larger(1)
    print *, 'check_is_larger(2):', check_is_larger(2)
    print *, 'check_is_larger(4):', check_is_larger(4)

    ! Placeholder for do_collapse testing
    print *, 'Testing do_collapse function (Placeholder)'

end program test_do_collapse

File: data/code_pairs/cpp/2314.cpp
#include <iostream>
#include <cassert>

bool check_is_larger(int i) {
    static int last_i = 0;
    bool is_larger = (i >= last_i) && ((i - last_i) <= 1);
    last_i = i;
    return is_larger;
}

void reset_check_is_larger() {
    check_is_larger(1); // Reset the function's static variable by simulating the initial call
}

void test_check_is_larger() {
    reset_check_is_larger(); // Ensure the function is reset before tests
    assert(check_is_larger(1) == true);
    assert(check_is_larger(2) == true);
    assert(check_is_larger(4) == false); // This will now match the Fortran output
}

// The do_collapse function is not detailed in prior context. Placeholder for testing.
int do_collapse() {
    // Placeholder implementation based on assumed logic. Original logic was not fully specified.
    return 0; // Placeholder return
}

void test_do_collapse() {
    assert(do_collapse() == 0); // Placeholder test
}

int main() {
    std::cout << "Testing check_is_larger function" << std::endl;
    test_check_is_larger();
    std::cout << "check_is_larger function tests passed" << std::endl;

    std::cout << "Testing do_collapse function (Placeholder)" << std::endl;
    test_do_collapse();
    std::cout << "do_collapse function tests passed (Placeholder)" << std::endl;

    return 0;
}